### DISCUSSION TITLE
Export to MusicXML

### DISCUSSION THREAD

**Discussion by danielgarthur** [neanes/neanes#332] [Ideas]
I want to be able to export to MusicXML, which can be imported into scorewriting software such as Finale or Musescore. The export should only approximate the microtonal intervals, using the 12-tone Western diatonic intervals common to Western staff notation.

**Specs**

- Ni = C
- Option to switch between using A or Ab in the soft chromatic with base on G
- No bar lines

**Complications**

- Transpositions created by fthores may be difficult to arbitrarily code
- How to encode embellishments? Should they simply be ignored?

**Comment by basil** [comment]
Here is my philosophy, which is by no means the only one, but certainly a practical one that is informed by many years of personal experience. It is quite close to the philosophy used by e.g. Fr Seraphim Dedes and quite far from the philosophy used by e.g. John Michael Boyer. My philosophy does make some assumptions about the skill level of the target audience that may not hold for professional musicians but often hold for amateur church musicians. This is quite deliberate, and I don't claim that my philosophy is the only one, just that it has worked well for me.

- Don't attempt to notate microtones. This is quite often a bad fit for the target audience. The target audience is more likely to learn these by ear. This might be the most controversial part of my philosophy, but it sounds like we are already in agreement about this with regard to the software.
- Use custom key signatures. For example, hard chromatic needs a custom key signature of E♭, B♭, C♯, and F♯. Attempts to notate this in D minor (B♭) or G minor (E♭ and B♭) result in unnecessary notational clutter. Finale and NoteWorthy Composer both allow custom key signatures to be defined. Although this is stretching Western notation a bit, in practice the target audience rarely has any issues with this.
- Notate ison as [chords](https://en.wikipedia.org/wiki/Chord_notation), with "Un." for "Unison". Attempts to create a second staff or layer within a single staff tend to clutter the score. Although this is stretching Western notation a bit, in practice the target audience rarely has any issues with this after a very brief explanation of what isokratima is.
- If the original piece has (Karas-style) barlines throughout the whole score (rather than just occasionally on deviations from duple meter, Simonopetra style), then by all means keep those barlines in the transcription. They are quite valuable when directing a choir or for people with an existing Western musical background. In Byzantine notation the accent-based rhythm is often visually discernible with the petaste and psifiston, but this is not as easily visually discernible in Western notation without barlines. However, one should dispense with the time signatures, which are big and bold and clutter the score. In Finale and NoteWorthy Composer you can easily hide time signatures. Papa Ephraim does not use barlines at all, while Fr Seraphim uses them. In practice the target audience often does better with the latter. Another benefit of using barlines is that it is very clear when accidentals terminate, which is far more ambiguous when no barlines are used.
- A one-beat petaste or psifiston that is commonly executed with a flutter (not all of them) should be notated with an [accent mark](https://en.wikipedia.org/wiki/Accent_(music)) rather than grace notes to reduce clutter. Papa Ephraim (over)does this, Fr Seraphim does not notate it at all. I think it adds a lot of richness to the result and people can be easily trained to execute it without much difficulty, just like (abusing) chord markings for isokratima.
- The score should not be completely without any embellishments at all, as dry metrophonia sounds horrible. However, a tasteful amount of interpretation should be introduced to avoid cluttering the score and potentially alienating significant portions of the target audience. Interpretation should be limited to less controversial practices that are common to multiple schools of Byzantine music rather than more controversial hallmarks of a particular school (e.g., Karas-style or Stanitsas-style). To avoid cluttering the score and making pieces too difficult to sight-read, analysis should focus on the level of dotted eighth notes and sixteenth notes rather than frequent 16th or 32nd notes. In practice the target audience often does better with Fr Seraphim's transcriptions than Papa Ephraim's. John Michael Boyer's transcriptions are often prohibitively difficult for the target audience. My personal transcription style would be somewhere between Fr Seraphim's and Papa Ephraim's.

---

I think the intermediate representation from #322 would be a good place to start. The analysis service could be improved to optionally apply some basic interpretation, as described above. This could be optionally present in playback as well as staff notation export. The intermediate representation, when this mode is enabled, would have some basic analysis. And this intermediate representation could be passed along to playback, (hypothetical) MIDI export, (hypothetical) staff notation export, etc. Once #322 goes in I would be interested in starting to work on the "interpretation" mode described above to improve the intermediate representation for both playback and this feature.

**Comment by Mykbibby** [comment]
Agree with everything here. This + the ability to import pre-existing Byzantine scores as a starting point are the key two features I'm looking forward to.

**Reply by Mykbibby** [reply]
Bumping this... Would love to see this supported in the near future.

**Reply by danielgarthur** [reply]
@Mykbibby Not sure if you saw the release, but I did add a first pass at MusicXML export. It works best for scores that contain explicit measure bars everywhere. Otherwise, you'll probably spend a while formatting the staff notation. I also noticed that Musescore seems to have the best MusicXML support of the scorewriting applications that I tried.
